array/DNAm/preprocessing/checkRconfigFile.r

```diff
@@ -21,144 +21,84 @@
-  if(!exists(i)){
-    stop(paste0("'", i, "' must be defined in the config file" ))
-  }
-}
-
-
-# if ctCheck is TRUE, check extra params and thresholds needed to run clusterCellTypes.r script
-
-if(ctCheck == TRUE){
-  for(i in c(ctCellParams, ctThres)){
-    if(!exists(i)){
-      stop(paste0("To run ctCheck'", i, "' must be defined in the config file" ))
+check_parameters <- function(parameters, type_check, warning_message) {
+  for (parameter in parameters) {
+    if (!exists(parameter)) {
+      bad_parameter_exists <- TRUE
+      warning("\n'", parameter, "' must be defined in the config file\n")
     }
-  }
-}
-
-
-# check calcQCmetrics.r script thresholds
-
-for(i in qcthres){
-    if(!is.numeric(get(i))){
-      stop(paste0("'", i, "' must be numeric"))
-  } 
-} 
-
-
-
-# check logical parameters
-
-for(i in logicalParams){
-    if(!is.logical(get(i))){
-      stop(paste0("'", i, "' must be True/False"))
-  } 
-} 
-
-                    
-
-# if ctCheck is TRUE, check extra params and thresholds needed to run clusterCellTypes.r script
-
-
-if(ctCheck == TRUE){
-  for(i in ctCellParams){
-    if(exists(i)){
-      if(!is.character(get(i))){
-        stop(paste0("'", i, "' must be string"))
-      }
-    } else {
-      stop(paste0("To run ctCheck '", i, "' must be defined in the config file" ))
+    if (!type_check(get(parameter))) {
+      bad_parameter_exists <- TRUE
+      warning("\n'", parameter, "' ", warning_message, "\n")
     }
+    message("'", parameter, "' is correctly defined.")
   }
-  for(i in ctThres){
-    if(exists(i)){
-      if(!is.numeric(get(i))){
-        stop(paste0("'", i, "' must be numeric"))
-      } 
-    } else {
-      stop(paste0("To run ctCheck '", i, "' must be defined in the config file" ))
-    }
-  } 
 }
 
+check_parameters(qcthres, is.numeric, "must be numeric")
+check_parameters(logicalParams, is.logical, "must be TRUE or FALSE")
 
-# check 'ungrouped' params are correctly formatted
-
-if(!toupper(tissueType) %in% c("BRAIN", "BLOOD")){
-  stop("Unrecognised tissueType. Must be either 'blood' or 'brain'")
+if (ctCheck) {
+  message(
+    "\n`ctCheck` is set to TRUE.",
+    "Additional parameters need to be checked...\n"
+  )
+  check_parameters(ctCellParams, is.character, "must be a string")
+  check_parameters(ctThres, is.numeric, "must be numeric")
 }
 
 
-if(!toupper(arrayType) %in% c("HM450K", "V1", "V2")){
-  stop("Unrecognised arrayType. Must be 'HM450K', 'V1' or 'V2'")
+if (!toupper(tissueType) %in% c("BRAIN", "BLOOD")) {
+  bad_parameter_exists <- TRUE
+  warning("\nUnrecognised tissueType. Must be either 'blood' or 'brain'\n")
+}
+if (!toupper(arrayType) %in% c("HM450K", "V1", "V2")) {
+  bad_parameter_exists <- TRUE
+  warning("\nUnrecognised arrayType. Must be 'HM450K', 'V1' or 'V2'\n")
 }
 
 
-for(i in c("Sentrix_ID", "Sentrix_Position")){
-  if(!i %in% techVar){
-    stop(paste0("'", i, "' must be included in techVar"))
+for (i in c("Individual_ID", "Cell_Type", "Sex")) {

```

Comment:
Technically I think the the scripts can handle bioVar and techVar not being defined (line 133 and 134 of QC.rmd: 

`if (!exists("techVar")) techVar <- NULL`
`if (!exists("bioVar")) bioVar <- NULL`

In practice. Sentrix_ID and Sentrix_Position are always going to need to be included in the QC. It would probably be better to hard code them in, but that's beyond the scope of this PR..